src/compiler/csharp_generator.cc

```diff
@@ -184,34 +174,29 @@ std::string GetServerClassName(const ServiceDescriptor* service) {
   return service->name() + "Base";
 }
 
-std::string GetCSharpMethodType(MethodType method_type) {
-  switch (method_type) {
-    case METHODTYPE_NO_STREAMING:
-      return "grpc::MethodType.Unary";
-    case METHODTYPE_CLIENT_STREAMING:
-      return "grpc::MethodType.ClientStreaming";
-    case METHODTYPE_SERVER_STREAMING:
-      return "grpc::MethodType.ServerStreaming";
-    case METHODTYPE_BIDI_STREAMING:
+std::string GetCSharpMethodType(const MethodDescriptor* method) {
+  if (method->client_streaming()) {
+    if (method->server_streaming()) {
```

Comment:
In cases where you have 4 possible outcomes, I think using `if () {}  else {}`  for both outer and inner `if` statements ends up being more readable than falling back to the "default"?
Also below for `GetCSharpServerMethodType`

Reply:
Updated